contracts/SportMarkets/Parlay/ParlayMarketsAMM.sol

```diff
@@ -0,0 +1,760 @@
+                inverseSum,
+                sUSDAfterFees,
+                inverseQuotes
+            );
+            (totalQuote, totalBuyAmount, finalQuotes, ) = _calculateFinalQuotes(_sportMarkets, _positions, amountsToBuy);
+            if (totalQuote > 0) {
+                uint expectedPayout = ((sUSDAfterFees * ONE * ONE) / totalQuote) / ONE;
+                skewImpact = expectedPayout > totalBuyAmount
+                    ? (((ONE * expectedPayout) - (ONE * totalBuyAmount)) / (totalBuyAmount))
+                    : (((ONE * totalBuyAmount) - (ONE * expectedPayout)) / (totalBuyAmount));
+                amountsToBuy = _applySkewImpactBatch(amountsToBuy, skewImpact, (expectedPayout > totalBuyAmount));
+                totalBuyAmount = _applySkewImpact(totalBuyAmount, skewImpact, (expectedPayout > totalBuyAmount));
+            }
+        }
+    }
+
+    function _applySkewImpact(
+        uint _value,
+        uint _skewImpact,
+        bool _addition
+    ) internal pure returns (uint newValue) {
+        newValue = _addition ? (((ONE + _skewImpact) * _value) / ONE) : (((ONE - _skewImpact) * _value) / ONE);
+    }
+
+    function _applySkewImpactBatch(
+        uint[] memory _values,
+        uint _skewImpact,
+        bool _addition
+    ) internal pure returns (uint[] memory newValues) {
+        uint totalAmount;
+        newValues = new uint[](_values.length);
+        for (uint i = 0; i < _values.length; i++) {
+            newValues[i] = _applySkewImpact(_values[i], _skewImpact, _addition);
+            totalAmount += newValues[i];
+        }
+    }
+
+    function _calculateFinalQuotes(
+        address[] memory _sportMarkets,
+        uint[] memory _positions,
+        uint[] memory _buyQuoteAmounts
+    )
+        internal
+        view
+        returns (
+            uint totalQuote,
+            uint totalBuyAmount,
+            uint[] memory finalQuotes,
+            uint[] memory buyAmountPerMarket
+        )
+    {
+        buyAmountPerMarket = new uint[](_sportMarkets.length);
+        finalQuotes = new uint[](_sportMarkets.length);
+        for (uint i = 0; i < _sportMarkets.length; i++) {
+            totalBuyAmount += _buyQuoteAmounts[i];
+            if (reducedFeesEnabled) {
+                buyAmountPerMarket[i] = sportsAmm.buyFromAmmQuoteForParlayAMM(
+                    _sportMarkets[i],
+                    _obtainSportsAMMPosition(_positions[i]),
+                    _buyQuoteAmounts[i]
+                );
+            } else {
+                buyAmountPerMarket[i] = sportsAmm.buyFromAmmQuote(
+                    _sportMarkets[i],
+                    _obtainSportsAMMPosition(_positions[i]),
+                    _buyQuoteAmounts[i]
+                );
+            }
+            if (buyAmountPerMarket[i] == 0) {
+                totalQuote = 0;
+                totalBuyAmount = 0;
+                break;
+            }
+        }
+        for (uint i = 0; i < _sportMarkets.length; i++) {
```

Comment:
because the `totalQuote` is recalculated based on the `finalQuotes`.
The `finalQuotes` are highly impacted by the `skewImpact` in the SportsAMM. 
Then later, the global parlay skewImpact is calculated, and the `totalQuote` is increased compared to the `initalQuote`.
Sometimes up to 25%